src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -290,52 +360,70 @@ private static <E extends Enum<E>> Object convertEnumConstants(MethodHandles.Loo
             }
             return label;
         } else if (labelClass == String.class) {
-            @SuppressWarnings("unchecked")
-            Class<E> enumClass = (Class<E>) enumClassTemplate;
-            try {
-                return ConstantBootstraps.enumConstant(lookup, (String) label, enumClass);
-            } catch (IllegalArgumentException ex) {
-                return null;
-            }
+            return EnumDesc.of(enumClassTemplate.describeConstable().get(), (String) label);
         } else {
             throw new IllegalArgumentException("label with illegal type found: " + labelClass +
                                                ", expected label of type either String or Class");
         }
     }
 
-    private static int doEnumSwitch(Enum<?> target, int startIndex, Object[] unresolvedLabels,
-                                    MethodHandles.Lookup lookup, Class<?> enumClass,
-                                    ResolvedEnumLabels resolvedLabels) {
-        Objects.checkIndex(startIndex, unresolvedLabels.length + 1);
+    private static <T extends Enum<T>> int mappedEnumLookup(T value, MethodHandles.Lookup lookup, Class<T> enumClass, EnumDesc<?>[] labels, EnumMap enumMap) {
+        if (enumMap.map == null) {
+            T[] constants = SharedSecrets.getJavaLangAccess().getEnumConstantsShared(enumClass);
+            int[] map = new int[constants.length];
+            int ordinal = 0;
 
-        if (target == null)
-            return -1;
+            for (T constant : constants) {
+                map[ordinal] = labels.length;
 
-        if (resolvedLabels.resolvedLabels == null) {
-            resolvedLabels.resolvedLabels = Stream.of(unresolvedLabels)
-                                                  .map(l -> convertEnumConstants(lookup, enumClass, l))
-                                                  .toArray();
+                for (int i = 0; i < labels.length; i++) {
+                    if (Objects.equals(labels[i].constantName(), constant.name())) {
+                        map[ordinal] = i;
+                        break;
+                    }
+                }
+
+                ordinal++;
+            }
         }
+        return enumMap.map[value.ordinal()];
```

Comment:
`enumMap.map` never gets set before this line.

Reply:
There's a bug filed for this already: https://bugs.openjdk.org/browse/JDK-8332522 

@lahodaj explained that this broken code is part of an optimization which is never attempted (IIRC due to the bug you noted on line 327). JDK-833522 seem like a good place to continue this conversation..?